constellation-app/constellation
# Fix Add hashmod action and added help

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Modified code to enable the csv file to be loaded into the graph.
Some logic changes and type changes had to be catered for.


### Alternate Designs
None considered.


### Why Should This Be In Core?
To stop the infinite loop on the action and to enable Add Hashmod to actually load the csv file into the graph.


### Benefits
To stop the infinite looping and possible crash and to enable Add Hashmod action.


### Possible Drawbacks
Add Hashmod action not working.


### Verification Process

1. Select Experimental -> Tools -> Add Hashmod
2. Click on "Single" button and select sample .csv file. Click Open.
3. Select options Create new Vertexes for non matching keys and Create new attributes from names in Column Headers.
4. View with Table View and Attribute Editor and see that all nodes and attributes have been added.



### Applicable Issues
#2147
